### ISSUE TITLE
Accessibility: review and make accessibility implementation consistent

### ISSUE THREAD

**Issue by M-Lipin** [dotnet/winforms#3287]
* .NET Core Version: 

.NET SDK (reflecting any global.json):
Version: 5.0.100-preview.5.20260.4
Commit: 0950d416f9

Runtime Environment:
OS Name: Windows
OS Version: 10.0.18363
OS Platform: Windows
RID: win10-x64
Base Path: C:\Program Files\dotnet\sdk\5.0.100-preview.5.20260.4\

Host (useful for support):
Version: 5.0.0-preview.5.20260.5
Commit: 09ee4814cb

* Have you experienced this same bug with .NET Framework?: 

Yes

**Problem description:**

```
object IAccessible.accHitTest(int xLeft, int yTop)
{
    // Some code is here...

    AccessibleObject obj = HitTest(xLeft, yTop);
    if (obj != null) // This is not consistent with below implementation.
    {
        return AsVariant(obj);
    }

    // Other statements here...
    // Execution goes here if obj is null. Do we need it?
}
```

```
object IAccessible.accNavigate(int navDir, object childID)
{
    // Some code is here...

    return AsVariant(child.Navigate((AccessibleNavigation)navDir));
}
```

**Expected behavior:**
Using `AsVariant` should be consistent across AccessibleObject implementation, if code flow is not consistent we should have a comment in the code clarifying the intent of different implementation.

**Minimal repro:**
Create the app with some control, use Inspect to force some accessibility calls. Observe different code flows for different methods.

**Comment by Tanya-Solyanik** [comment]
This code had been cleaned up when moving to the cswin32
